### ISSUE TITLE
M³ crashing

### ISSUE THREAD

**Issue by khcarmel** [sircharlo/meeting-media-manager#260]
The issue that we are experiencing is that when we launch JWMMF along with Zoom JWMMF closes unexpectedly. We are in a doubleKingdom Hall with multiple computers. The issue happens with some of the computers but not with all. But we have at least three that we are experiencing the issue. If JWMMF is opened without zoom it is stable, however once zoom is opened on the PC it closes. Regardless of the combination opening zoom 1st or second the same result happens.

https://user-images.githubusercontent.com/101886948/164007748-f57956e7-68e6-4e87-9bcc-7317b75de629.mov

.

**Comment by sircharlo** [comment]
Hi, thanks for the video! 
I don't like that it's not reproducible on all computers; those bugs are usually the most difficult to find 😆. 
Is there anything different on the computers that are experiencing this issue?

In any case:
- Does this happen only if JWMMF is in media presentation mode?
- Can you try wiping Zoom completely using Zoom's tool [here](https://assets.zoom.us/docs/msi-templates/CleanZoom.zip), and then reinstalling Zoom and seeing if the issue persists with JWMMF?

**Comment by khcarmel** [comment]
Hi, yes that seems to be the case when JWMMF is in the media presentation mode the closure takes place. Now with regards to computers at the hall two of the computers are Dell Alienware PCs and the issue is happening there. On one of our backup laptops the issue is happening there as well. There are a total of six PCs associated with the hall as mentioned before is happening on three of them. I did uninstall and reinstall zoom several times but I did not utilize the tools that you provided, so I will give that a try. Also the computers are all configured the same way.

**Comment by sircharlo** [comment]
Sounds good; let me know if wiping Zoom helps. If not, I'll give you a beta version of JWMMF that was crash reports enabled. That will help me to target the issue, if you're willing to help.

**Comment by sircharlo** [comment]
Additionally, is Zoom set-up to work with dual monitors as per [this link](https://support.zoom.us/hc/en-us/articles/201362583-Using-Zoom-desktop-client-with-dual-monitors)?

- If so, try disabling it and restarting the computer
- If not, try enabling it and restarting the computer

**Comment by khcarmel** [comment]
So I’m here at the hall now, and I removed zoom by means of the link that you provided. I then reinstalled it After rebooting the computer and  the same behavior happened.

Thanks Shawn

**Comment by sircharlo** [comment]
> Additionally, is Zoom set-up to work with dual monitors as per [this link](https://support.zoom.us/hc/en-us/articles/201362583-Using-Zoom-desktop-client-with-dual-monitors)?
> 
> * If so, try disabling it and restarting the computer
> * If not, try enabling it and restarting the computer

What about this?

**Comment by khcarmel** [comment]
Yes this was done quite some time ago duel monitors are set up correctly

Thanks

**Comment by sircharlo** [comment]
This to me sounds like it's a problem either with Zoom or with the display drivers on that specific hardware. 

1. Try installing [this custom version](https://we.tl/t-VIeU1wpsyj) of JWMMF and let me know if it still crashes. In that version, JWMMF won't attempt to use your display drivers at all.
2. Whether that works well or not, try to install [this version](https://we.tl/t-nlKoUpaVtN) afterwards. It should crash just like the official version of JWMMF, except that a crash report file will be generated when that happens, which will help me to debug some more. The crash file should be located under `C:\Users\[Username]\AppData\Roaming\jw-meeting-media-fetcher\Crashpad\reports`.

**Comment by khcarmel** [comment]
OK, I will try this tomorrow and will report back to you my findings. Thanks so very much for your assistance.

**Comment by khcarmel** [comment]
Sure thing, I would be happy to

**Comment by khcarmel** [comment]
Thanks Shawn

**Comment by Chipperchap** [comment]
No possibility that antivirus McAfee is interfering? I just see it on your desktop in the video and wondered...

**Comment by khcarmel** [comment]
Thanks for the inquiry, no I believe McAfee has nothing to do with the issue. As JWMMF was working for our first and second in person meeting. However it is not working now with no changes to McAfee. Unfortunately there was an upgrade to both JWMMF as well as zoom wiith the update to JWMMF being first.

**Comment by sircharlo** [comment]
Notably, a Zoom update was released this week, version 5.10.3 (4851). Perhaps this update fixes something as well?

**Comment by khcarmel** [comment]
Tried installing the new zoom update. The results are the same. Moving on to next stage

Thanks Shawn

**Comment by khcarmel** [comment]
So after installing JWMMF the version that does not used to display driver JWMMF closed right after opening zoom. I am now moving on to the second version of JWMMF that was sent. By deleting the Second version provided and installing the third version With debug.

**Comment by khcarmel** [comment]
The Debug version of JWMMF is now installed. It took a little while longer for JWMMF to close but it eventually did. Around 15 seconds later. I also check the Zoom settings and other than mirroring the settings you provided are exactly the same.

**Comment by sircharlo** [comment]
Can you send me the debug files found in `C:\Users\[Username]\AppData\Roaming\jw-meeting-media-fetcher\Crashpad\reports` please?

**Comment by sircharlo** [comment]
Also, does the issue happen if you exit fullscreen on the JWMMF black screen that goes on the TVs (by clicking on the yeartext and hitting <kbd>F11</kbd>), and drag the yeartext to your main (soundbooth) screen?

**Comment by sircharlo** [comment]
And finally, does the issue happen if you **disable** dual monitor mode in Zoom?

**Comment by khcarmel** [comment]
I’m not able to locate the appdata folder under user on the c drive

Thanks Shawn

**Comment by sircharlo** [comment]
> I’m not able to locate the appdata folder under user on the c drive

Hit <kbd>Windows</kbd> <kbd>R</kbd>, type in `%localappdata%` and press <kbd>Enter</kbd>. You should be ble to navigate to it from there.

**Comment by khcarmel** [comment]
I have just tried disableing  duel monitors with same results

Thanks Shawn

**Comment by sircharlo** [comment]
> > I’m not able to locate the appdata folder under user on the c drive
> 
> Hit Windows R, type in `%localappdata%` and press Enter. You should be ble to navigate to it from there.

I apologize, I wrote too fast! Try hitting Windows R, type in `%appdata%` (no "local" needed), and press Enter.

**Comment by sircharlo** [comment]
> I have just tried disableing duel monitors with same results Thanks Shawn
> […](#)
> On Apr 20, 2022, at 1:44 PM, Charles Olivier Savignac ***@***.***> wrote: ﻿ I’m not able to locate the appdata folder under user on the c drive Hit Windows R, type in %localappdata% and press Enter. You should be ble to navigate to it from there. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.

I appreciate you trying all these things! I'm sorry this bug is proving to be quite pesky.

Could you also try disabling hardware acceleration in Zoom (all 3 checkboxes highlighted in red from the screenshot above) and seeing if that makes a difference?

**Comment by khcarmel** [comment]
Just to close the circle I did disable all three choices in zoom with the same results

Sorry I thought I sent this.

**Comment by sircharlo** [comment]
> > > I’m not able to locate the appdata folder under user on the c drive
> > 
> > 
> > Hit Windows R, type in `%localappdata%` and press Enter. You should be ble to navigate to it from there.
> 
> I apologize, I wrote too fast! Try hitting Windows R, type in `%appdata%` (no "local" needed), and press Enter.

Ok; did you find the crash file from here? 
`C:\Users\[Username]\AppData\Roaming\jw-meeting-media-fetcher\Crashpad\reports`

**Comment by khcarmel** [comment]
The zip contact is what I found in the folder and I forwarded it to you.

Did you not receive it?

**Comment by sircharlo** [comment]
No i didn't see anything. Can you try posting it as a comment here?

**Comment by khcarmel** [comment]
[99b76acb-1c20-49b2-a0a9-c957b5be374f.zip](https://github.com/sircharlo/jw-meeting-media-fetcher/files/8524665/99b76acb-1c20-49b2-a0a9-c957b5be374f.zip)

**Comment by khcarmel** [comment]
I hope you received it. If not I will have to send it to you this evening as I had to leave the Kingdom Hall. My apologies I should have double checked before leaving.

Shawn

**Comment by sircharlo** [comment]
I did receive it! I'll do the analysis now. Thanks so much for your time.

**Comment by sircharlo** [comment]
So, the error code thrown by JWMMF as it crashes is `0x80000003`. This particular code usually indicates a problem with the registry, a corrupted system file, or a memory issue. In any case, the problem seems to be independent from Zoom or JWMMF.

Here are the recommended steps to try and fix that error code, should you want to try.

1. **Run System File Checker (SFC) Scan** 
     Open an elevated command prompt. To do that open a Run dialog by pressing Win + R. Type cmd and press Ctrl + Shift + Enter to launch the    command prompt with administrative privileges. 
    Copy and paste the following commands, pressing Enter after each one:	
	 
    - `Dism /Online /Cleanup-Image /CheckHealth` 		 
    - `Dism /Online /Cleanup-Image /ScanHealth` 		 
    - `Dism /Online /Cleanup-Image /RestoreHealth 		 
    sfc /scannow`
2. **Run Disk Checker.**
    Open an elevated command prompt.
    Type the following command and press Enter to launch chkdsk:
    `chkdsk C: /r`
    Once the scan finishes restart your PC.
3. **Check For Memory Leaks.**
     Open a Run window by pressing Win + R. 
     Type mdsched.exe and press Enter to launch Memory Diagnostic Tool at the next startup. 
     Now restart your PC using the prompt that appears.
    After a reboot, perform a basic scan or go for the ‘Advanced’ options like ‘Test mix’ or ‘Pass count’. Hit F10 to start the test.
4. **Fix registry errors**
     Install CCleaner. Scan and fix registry errors using the Registry tab.

**Comment by khcarmel** [comment]
Thanks so very much for your time energy and effort. I will try to get this accomplished today as we look forward to using the JWMMF program along with zoom. No as a matter of figuring out how the issue arose in the first place. We have a lot of chefs in the soup and sometimes that makes a mess.

Thanks Shawn

**Comment by khcarmel** [comment]
I just wanted to provide an update in regards to our issue. I tried implementing all of the fixes that you provided to no avail. JWMMF closes immediately upon opening Zoom. But here’s a little wrinkle the profile that I was originally using did not have administrative rights meaning it set up as a secondary user. When I logged in as the administrator and installed JWMMF remain stable and I’m able to use it. Do you know of a way of comparing the two as they reside on the same PC. I also noticed that when I tried to delete JWMMF and reinstall it  all of my inputs, that is my congregation names, times all remain. So I’m wondering if I did a complete delete could that resolve my issue. 

Thanks Shawn

**Comment by sircharlo** [comment]
Hm, that is interesting. Was Zoom installed for all users (as admin, from MSI package), or is it installed using the local non-admin installer for each user? 
To wipe all of JWMMF's data, including saved settings, delete the following folder: `C:\Users\[Username]\AppData\Roaming\jw-meeting-media-fetcher`

**Comment by sircharlo** [comment]
It seems you're not the only one experiencing [this issue](https://github.com/electron/electron/issues/33862).
We'll have to wait and see if it gets resolved with a future Electron update, I suppose.

**Comment by khcarmel** [comment]
Good morning,

Let me begin by thanking you so very much for all of your hard work M3. This past weekend I have installed the upgraded program on all of our computers and it is working as expected. I will tell you that we did have a resolve using JWMMF after eliminating stream deck software. I don't know if this was resolved or not however they were stable for two or three weeks. After installing in M3 (appreciate all the updates) are meeting over the weekend all worked without any issues.
